workflows/gasclustering.nf

```diff
@@ -47,17 +48,58 @@ include { CUSTOM_DUMPSOFTWAREVERSIONS } from '../modules/nf-core/custom/dumpsoft
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 */
 
+
+def validateFilePath(String filep){
+    // TODO request clarification if there is a better way to perform this operation
+    if(filep){
+        file_in = file(filep)
+        if(file_in.exists()){
+            return file_in
+        }else{
+            exit 1, "${filep} does not exist but was passed to the pipeline. Exiting now."
+        }
+    }
+    return [] // empty value if file argument is null
+}
+
 workflow GASCLUSTERING {
 
     ch_versions = Channel.empty()
 
     // Create a new channel of metadata from a sample sheet
     // NB: `input` corresponds to `params.input` and associated sample sheet schema
-    input = Channel.fromSamplesheet("input").map {meta, file -> file }.collect()
+    input = Channel.fromSamplesheet("input")
+    merged_input = input.map{
+        meta, mlst_files -> mlst_files
+    }.collect()
+
+    merged = LOCIDEX_MERGE(merged_input)
+    ch_versions = ch_versions.mix(merged.versions)
+
+    // optional files passed in
+    mapping_file = validateFilePath(params.pd_mapping_file)
+    columns_file = validateFilePath(params.pd_columns)
```

Comment:
As above, I would expect `true` or `false` to return from a function like this.

Reply:
Would a good fix to this be simply changing the function name to better reflect what the function is doing?

Reply:
I guess if the function was called something like `prepareFilePath`?

What comes to mind is there are functions built into programming languages that will throw FileNotFound exceptions in exceptional circumstances, which is considered fine behaviour, but I'm wondering if a `prepareFilePath` method is an exceptional circumstance? I'm not sure.

Reply:
Ill refactor and re-name the function and hopefully that will help with things, I think part of what makes this function so odd is returning the `[]` letting the empty array act as an empty value that is still 'valid' as an input for nextflow processes.

Reply:
Renamed the function and altered its return types!